### PR TITLE
Stability: Fix Dynamic Forward Proxy Test Flake

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Fix a flake in the DFP e2e test

# Context

When the client timeout < server response time, we will encounter `Client.Timeout exceeded while awaiting headers`. The solution here is to bump the timeout of our test client.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/8019

### CODE REVIEW

File: test/testutils/http_client.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 // DefaultHttpClient should be used in tests because it configures a timeout which the http.DefaultClient
 // does not have
 var DefaultHttpClient = &http.Client{
-	Timeout: time.Second * 1,
+	Timeout: time.Second * 2,
```
[the remark this answers is not in the document]
Comment (elcasteel) [reply test/testutils/http_client.go:17]: +100000

[the remark this answers is not in the document]
Comment (sam-heilbron) [reply test/testutils/http_client.go:22]: good call! https://github.com/solo-io/gloo/pull/8028/commits/4760b998724e2b742a8601991205ba052317c7b1

**Review by elcasteel — APPROVED** [review]

**Review by kdorosh — DISMISSED** [review]

File: changelog/v1.14.0-beta18/dfp-e2e-flake.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+changelog:
+  - type: NON_USER_FACING
+    issueLink: https://github.com/solo-io/gloo/issues/8019
+    resolvesIssue: true
+    description: |
```
Comment (kdorosh) [inline changelog/v1.14.0-beta18/dfp-e2e-flake.yaml:5]: nit; i slightly prefer
```suggestion
    description: >-
```

as it renders better in the changelog bot logs. see https://yaml-multiline.info/ for more

**Review by jackstine — APPROVED** [review]
I am good accepting this, but I do not have a Linux machine set up to test this quickly.

**Review by jackstine — APPROVED** [review]

**Review by elcasteel — DISMISSED** [review]

**Review by kdorosh — APPROVED** [review]

**Review by elcasteel — APPROVED** [review]

**Review by kdorosh — APPROVED** [review]

**Pull request merged**